File: backend/src/trade_alpha/dao/account_config.py
"""AccountConfig Document model."""

from datetime import datetime
from typing import Optional
from pydantic import Field
from beanie import Document, PydanticObjectId


class AccountConfig(Document):
    """Account config document for MongoDB."""

    name: str
    initial_capital: float
    buy_fee_rate: float = Field(default=0.0003)
    sell_fee_rate: float = Field(default=0.0003)
    stamp_tax_rate: float = Field(default=0.001)
    min_fee: float = Field(default=5.0)
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Settings:
        collection = "account_configs"
        indexes = [
            "name",
        ]

File: frontend/src/api/accountConfig.ts
import api from './index'

export interface AccountConfig {
  id: string
  name: string
  initial_capital: number
  cash: number
  position: number
  buy_fee_rate: number
  sell_fee_rate: number
  stamp_tax_rate: number
  min_fee: number
  created_at: string
  updated_at?: string
}

export const accountConfigApi = {
  list: () => api.get<AccountConfig[]>('/account-configs'),
  get: (id: string) => api.get<AccountConfig>(`/account-configs/${id}`),
  create: (data: Partial<AccountConfig>) => api.post<AccountConfig>('/account-configs', data),
  update: (id: string, data: Partial<AccountConfig>) => api.put(`/account-configs/${id}`, data),
  delete: (id: string) => api.delete(`/account-configs/${id}`),
}
